contrib/endpoints/src/api_manager/context/service_context.cc

```diff
@@ -76,6 +76,18 @@ ServiceContext::ServiceContext(std::unique_ptr<ApiManagerEnvInterface> env,
 
   service_account_token_.SetAudience(
       auth::ServiceAccountToken::JWT_TOKEN_FOR_FIREBASE, kFirebaseAudience);
+
+  if (config_->server_config() &&
+      !config_->server_config()
+           ->api_check_security_rules_config()
+           .authorization_service_audience()
+           .empty()) {
+    service_account_token_.SetAudience(
+        auth::ServiceAccountToken::JWT_TOKEN_FOR_AUTHORIZATION_SERVICE,
```

Comment:
As we discussed, I think it makes sense to ask API producer to enter the accepted audience as a http.invoke() function parameter, instead of having it statically in ESP config. There are two advantages. First, the rules are allowed to call multiple http endpoints, each of which accepts different JWT audience. Second, ESP service config is not bound to rules logic. API producer can change the rules to call a different http endpoint without the need to change ESP config and redeploy.